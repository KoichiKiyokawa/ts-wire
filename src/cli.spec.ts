import { execSync } from 'child_process'
import fs from 'fs'
import { describe, expect, it, beforeEach, afterEach } from 'vitest'

function runCli(arg: string): string {
  return execSync(`tsx src/cli.ts ${arg}`).toString()
}

describe('cli help command snapshot test', () => {
  it.each(['--help', '-h', 'init --help', 'init -h'])('%s', (arg) => {
    expect(runCli(arg)).toMatchSnapshot()
  })
})

describe('init command snapshot test', () => {
  beforeEach(() => {
    fs.mkdirSync('tmp/src', { recursive: true })
  })

  afterEach(() => {
    fs.rmSync('tmp', { recursive: true })
  })

  it('normal', () => {
    expect(runCli('init -p tmp/src/wire.ts')).toMatchSnapshot()
    expect(fs.readFileSync('tmp/src/wire.ts', 'utf-8')).toMatchSnapshot()
  })
})
