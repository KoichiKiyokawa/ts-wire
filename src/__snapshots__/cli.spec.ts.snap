// Vitest Snapshot v1

exports[`cli help command snapshot test > --help 1`] = `
"cli.ts

resolve dependencies and generate

Commands:
  cli.ts init  make a wire.ts
  cli.ts       resolve dependencies and generate                       [default]

Options:
  -p, --project  Path to the tsconfig file          [default: \\"./tsconfig.json\\"]
  -i, --input    Path to the file that has providers    [default: \\"src/wire.ts\\"]
  -o, --output   Path to the output file      [default: \\"src/wire-generated.ts\\"]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;

exports[`cli help command snapshot test > -h 1`] = `
"cli.ts

resolve dependencies and generate

Commands:
  cli.ts init  make a wire.ts
  cli.ts       resolve dependencies and generate                       [default]

Options:
  -p, --project  Path to the tsconfig file          [default: \\"./tsconfig.json\\"]
  -i, --input    Path to the file that has providers    [default: \\"src/wire.ts\\"]
  -o, --output   Path to the output file      [default: \\"src/wire-generated.ts\\"]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;

exports[`cli help command snapshot test > init --help 1`] = `
"cli.ts init

make a wire.ts

Options:
  -p, --path     Path to the creating file              [default: \\"src/wire.ts\\"]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;

exports[`cli help command snapshot test > init -h 1`] = `
"cli.ts init

make a wire.ts

Options:
  -p, --path     Path to the creating file              [default: \\"src/wire.ts\\"]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;
